algorithms/kernel/svm/svm_train_thunder_impl.i

```diff
@@ -222,19 +221,17 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::SMO
     {
         SafeStatus safeStat;
         /* Gather data to local buffers */
-        const size_t nBlocks = nWS / _blockSizeWS;
+        const size_t blockSizeWS = services::internal::min<cpu, algorithmFPType>(nWS, 16);
+        const size_t nBlocks     = nWS / blockSizeWS;
         daal::threader_for(nBlocks, nBlocks, [&](const size_t iBlock) {
-            const size_t startRow = iBlock * _blockSizeWS;
+            const size_t startRow = iBlock * blockSizeWS;
 
             PRAGMA_IVDEP
             PRAGMA_VECTOR_ALWAYS
-            for (size_t i = startRow; i < startRow + _blockSizeWS; ++i)
+            for (size_t i = startRow; i < startRow + blockSizeWS; ++i)
             {
-                const size_t wsIndex = wsIndices[i];
-
-                ReadColumns<algorithmFPType, cpu> mtKernel(kernelWS.get(), i, 0, nVectors);
-                DAAL_CHECK_BLOCK_STATUS_THR(mtKernel);
-                const algorithmFPType * const kernelWSData = mtKernel.get();
+                const size_t wsIndex                       = wsIndices[i];
+                const algorithmFPType * const kernelWSData = kernelWS[i];
                 yLocal[i]                                  = y[wsIndex];
                 gradLocal[i]                               = grad[wsIndex];
                 oldAlphaLocal[i]                           = alpha[wsIndex];
```

Comment:
And you probably forgot `PRAGMA_IVDEP; PRAGMA_VECTOR_ALWAYS` before `for` at line 245

Reply:
it is not necessary. I will delete in the master.